File: src/main/java/iut/sae/algo/Algo.py

class Algo(AlgoException) :
    def RLE(in_str) : 
        # Provide your algo here
        res = "" 
        count = 1  
        for i in range(len(in_str) - 1) :
            if in_str[i] == in_str[i+1] : 
                count+=1
            if count == 10 : 
                res += "9" + in_str[i] 
                count = 1 
            else : 
                res+= str(count) + in_str[i] 
                count = 1 
        rle += str(count) + in_str[-1]
        return res 
    

    def RLE(in_str, iteration)  :  
        # Provide your algo here
        return "NotYetImplemented"
                                                      

    def unRLE(in_str)  :
        # Provide your algo here
        res = "" 
        i = 0 
        while i < len(in_str) :
            if in_str[i].isdigit() :
                cpt = int(in_str[i])
                i += 1
                res += in_str[i] * cpt
            else :
                res += in_str[i]
            i += 1 
        return res 

    def unRLE(in_str, iteration)  : 
        # Provide your algo here
        return "NotYetImplemented"

File: src/main/java/iut/sae/algo/Algo.java
package iut.sae.algo;


public class Algo{
    public static String RLE(String in){
        // Provide your algo here
        StringBuffer rle = new StringBuffer() ; 
        char[] tab = in.toCharArray() ;
        int cpt = 1;
        int i = 0 ;
        while (i < in.length() - 1) {
            if (tab[i] == tab[i + 1]) {
                cpt++ ; 
                if (cpt == 10) {
                    rle.append(9) ; 
                    rle.append(tab[i]); 
                    cpt = 1 ;
                }
            } else {
                rle.append(cpt) ;
                rle.append(tab[i]) ; 
                cpt = 1 ;
            } 
           i++ ; 
        }
        rle.append(cpt) ;
        rle.append(tab[i]) ; 
        return rle.toString();
    }

    public static String RLE(String in, int iteration) throws AlgoException{
        // Provide your algo here
        return "NotYetImplemented";
    }

    public static String unRLE(String in) throws AlgoException{
        // Provide your algo here
        return "NotYetImplemented";

    }

    public static String unRLE(String in, int iteration) throws AlgoException{
        // Provide your algo here
        return "NotYetImplemented";

    }
}

